lib/Sema/CSDiagnostics.cpp

```diff
@@ -101,6 +101,52 @@ Expr *FailureDiagnostic::getArgumentExprFor(Expr *anchor) const {
   return nullptr;
 }
 
+ConstraintLocator *
+FailureDiagnostic::getConstraintLocatorFor(Expr *expr) const {
+  auto locator = CS.getConstraintLocator(expr);
+
+  if (isa<DeclRefExpr>(expr)) {
+    locator = CS.getConstraintLocator(expr);
+  } else if (isa<UnresolvedDeclRefExpr>(expr)) {
+    locator = CS.getConstraintLocator(expr);
```

Comment:
FYI `UnresolvedDecltRefExpr` is eliminated by `PreCheckExpr` using `TypeChecker::resolveDeclRefExpr` and going to be replaced with either `DeclRefExpr`, `OverloadedDeclRefExpr` or `UnresolvedDotExpr` so there is no need to cover it here.